### ISSUE TITLE
Zitadel.gRPC - Multi-target NetStandard 2.1

### ISSUE THREAD

**Issue by Foxtrek64** [zitadel/zitadel-net#175]
In order to support multiple C# runtimes, multitargeting for NetStandard 2.1 should be added to the Zitadel.gRPC library.

Specifically, this will assist with the development of LuzFaltex.Zitadel's console library which allows desktop services to integrate with Zitadel's rest API.

## API change
Modify `config.target`'s target framework selection.
```diff xml
<PropertyGroup>
-     <TargetFramework>net6.0</TargetFramework>
+     <TargetFrameworks>netstandard2.1;net6.0</TargetFrameworks>
</PropertyGroup>
```

## Risks
I have examined the `Zitadel.gRPC` dependencies to ensure there are no issues, however a similar check has yet to be performed for Zitadel.Net library. It is possible that some third party nuget dependencies will not be compatible with NetStandard 2.1. That said, this compatibility is only necessary for the gRPC package, so if there are any issues, an alternative solution may be to modify that projects csproj instead to minimize the potential impact of this change.

---

Edit: If this change is approved, I am willing to PR the issue.

**Comment by fforootd** [comment]
Let me check this with @buehler and @hifabienne 👍

**Comment by buehler** [comment]
Hey @Foxtrek64

Sadly, I cannot second this request. Using `netstandard2.0` or `netstandard2.1` are legacy now and are no longer maintained. .NET Core 3.1 is out of support as well. The plan that Microsoft stated is to further streamline the .NET versions. Right now, `.NET 6.0` is the new long term support version.

Supporting netstandard is possible in the gRPC library (if it compiles correctly), but not advisable since there is no further support for netstandard. netstandard is combined in the `netX.0` versions.

As for the other library, it is not feasible to support the old standard since some core library functions depend on modern code. As far as I know, it is not possible to support `netstandard` in the web SDK library (which the other lib is).

To summarize: If absolutely needed, we could add `netstandard` to the grpc library, but there is no support whatsoever since these frameworks were dropped by Microsoft itself. The extension library ("Zitadel") cannot support `netstandard` because of code and framework dependencies. I'd strongly advice you to support `net6.0` and onwards (and ignore `netstandard` entirely) in your library as well. Modern desktop applications and REST applications should target `net6.0` and no other versions since all the support is dropped for older versions.

Regards

Sources:
- https://devblogs.microsoft.com/dotnet/the-future-of-net-standard/
- https://andrewlock.net/stop-lying-about-netstandard-2-support/

**Comment by Foxtrek64** [comment]
Thanks for the links! Andrew Lock's article definitely had some things I had not expected to find in it. I've removed the NetStandard target from LuzFaltex.Zitadel and will instead target Net6 and forward.
